dj/api/nodes.py

```diff
@@ -50,3 +71,144 @@ def read_nodes(*, session: Session = Depends(get_session)) -> List[NodeMetadata]
     List the available nodes.
     """
     return session.exec(select(Node)).all()
+
+
+@router.post("/nodes/", response_model=Union[ExistingSourceNode, ExistingNode])
+def create_node(
+    data: Union[SourceNodeCreator, NodeCreator],
+    session: Session = Depends(get_session),
+) -> Union[ExistingSourceNode, ExistingNode]:
+    """
+    Create a node
+    """
+    dj_settings = get_settings()
+    # Check if the node already exists
+    query = select(Node).where(Node.name == data.name)
+    node = session.exec(query).one_or_none()
+    if node:
+        raise HTTPException(
+            status_code=HTTPStatus.CONFLICT,
+            detail=f"Node already exists: {data.name}",
+        )
+
+    if data.type == NodeType.SOURCE:
+        return create_source_node(session=session, data=data)
+
+    new_node = Node.from_orm(data)
+    new_nodes_parents = get_dependencies(new_node.query) if new_node.query else set()
+
+    # Check the columns for any dimension references and do a metadata lookup of the dimension node
+    for column in new_node.columns:
+        if dimension_target := column.get("dimension"):
+            dimension_target, dimension_column = dimension_target.rsplit(".", 1)
```

Comment:
I wonder if we should use a different separator when indicating dimension, eg:

```yaml
dimension = foo.bar::column
```

Reply:
Ooh I really like that! I wonder if that would change anything about how the the parsing or converting the parse tree to a DJ AST works.

Reply:
This syntax (::) reminds me of the Postgres notation for casting a value to a type. What exactly do we want to represent here?

Reply:
Ah that's right, I knew that looked familiar. 😄

Reply:
Doesn't have to be `::`, could be `:`?

Reply:
`foo.bar->column`
`foo.bar:column`